src/main/cluster-manager.ts

```diff
@@ -118,10 +121,7 @@ export class ClusterManager extends Singleton {
         cluster.kubeConfigPath = entity.spec.kubeconfigPath;
         cluster.contextName = entity.spec.kubeconfigContext;
 
-        entity.status = {
-          phase: cluster.disconnected ? "disconnected" : "connected",
-          active: !cluster.disconnected
-        };
+        this.updateEntityStatus(entity, cluster);
```

Comment:
Wouldn't it be better to pass through `cluster.failureReason` too. That way `cluster-status` doesn't need to access `ClusterStore` at all.

Reply:
Maybe, I'll leave that to a separate PR.

Reply:
Sounds good